File: constants/typography.ts
import { Platform, type TextStyle } from 'react-native';

export const Typography = {
  display:  { fontSize: 34, fontWeight: '700', lineHeight: 41 },
  title1:   { fontSize: 28, fontWeight: '700', lineHeight: 34 },
  title2:   { fontSize: 22, fontWeight: '600', lineHeight: 28 },
  title3:   { fontSize: 20, fontWeight: '600', lineHeight: 25 },
  body:     { fontSize: 17, fontWeight: '400', lineHeight: 22 },
  callout:  { fontSize: 16, fontWeight: '400', lineHeight: 21 },
  subhead:  { fontSize: 15, fontWeight: '400', lineHeight: 20 },
  footnote: { fontSize: 13, fontWeight: '400', lineHeight: 18 },
  caption:  { fontSize: 12, fontWeight: '400', lineHeight: 16 },
  mono:     { fontSize: 15, fontFamily: Platform.OS === 'ios' ? 'Menlo' : 'monospace' },
} as const satisfies Record<string, TextStyle>;

export type TypographyToken = keyof typeof Typography;

File: HomesplitIOS/DesignSystem/Typography.swift
import SwiftUI

/// Typography scale parity with RN. Prefer the Dynamic Type-friendly system
/// styles where they map cleanly (`.body`, `.title2`, …); use these tokens
/// when matching the RN spec exactly is more important than scaling.
enum HSFont {
    static let display  = Font.system(size: 34, weight: .bold)
    static let title1   = Font.system(size: 28, weight: .bold)
    static let title2   = Font.system(size: 22, weight: .semibold)
    static let title3   = Font.system(size: 20, weight: .semibold)
    static let body     = Font.system(size: 17, weight: .regular)
    static let callout  = Font.system(size: 16, weight: .regular)
    static let subhead  = Font.system(size: 15, weight: .regular)
    static let footnote = Font.system(size: 13, weight: .regular)
    static let caption  = Font.system(size: 12, weight: .regular)
    static let mono     = Font.system(size: 15, design: .monospaced)
}
